### PR TITLE
DBZ-1760 Allow skipping of invalid events

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-1760

Use diff whithout whitespaces

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java [hunk 1]
```diff
@@ -79,6 +79,60 @@ public static Version parse(String value, String defaultValue) {
...
+     */
+    public static enum EventProcessingFailureHandlingMode implements EnumeratedValue {
+
+        /**
+         * Problematic events will be skipped.
+         */
+        IGNORE("ignore"),
+
+        /**
+         * Problematic event and will be logged and the events will be skipped.
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java]:
```suggestion
         * Problematic events will be logged and skipped.
```

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/EventProcessingFailureHandlingIT.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import io.debezium.config.CommonConnectorConfig.EventProcessingFailureHandlingMode;
+import io.debezium.config.Configuration;
+import io.debezium.connector.sqlserver.SqlServerConnectorConfig.SnapshotMode;
+import io.debezium.connector.sqlserver.util.TestHelper;
+import io.debezium.embedded.AbstractConnectorTest;
+import io.debezium.junit.logging.LogInterceptor;
+import io.debezium.util.Testing;
+
+/**
+ * Integration test for the Debezium SQL Server connector.
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/EventProcessingFailureHandlingIT.java]: Something more specific would be nice.

File: debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java [hunk 3]
```diff
@@ -79,6 +79,60 @@ public static Version parse(String value, String defaultValue) {
...
 
+    /**
+     * The set of predefined modes for dealing with failures during event processing.
+     */
+    public static enum EventProcessingFailureHandlingMode implements EnumeratedValue {
+
+        /**
+         * Problematic events will be skipped.
+         */
+        IGNORE("ignore"),
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java]: Perhaps `SKIP` is more expressive.

File: debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java [hunk 4]
```diff
@@ -130,43 +132,65 @@ public SnapshotReceiver getSnapshotChangeEventReceiver() {
...
+                    changeRecordEmitter.getOffset().getOffset(),
+                    this::enqueueHeartbeat);
 
-        return handled;
+            return handled;
+        }
+        catch (Exception e) {
+            switch (connectorConfig.getEventProcessingFailureHandlingMode()) {
+                case FAIL:
+                    LOGGER.error(
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java]: Either log or throw, not both.

File: documentation/modules/ROOT/pages/connectors/db2.adoc [hunk 5]
```diff
@@ -1446,6 +1446,13 @@ twice - once in initial snapshot and once in streaming phase. Nonetheless, that
 data mirroring.
 For `read_uncommitted` there are no data consistency guarantees at all (some data might be lost or corrupted).
 
+|`event.processing{zwsp}.failure.handling.mode`
+|`fail`
+| Specifies how the connector should react to exceptions during processing of events.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/db2.adoc]: > indicating the problematic event and its offset

Actually only the offset is revealed, which I think is good. So the comment should be adjusted. Same for the enum JavaDoc.

**Comment by jpechane** [comment]
@gunnarmorling All comments should be addressed

**Comment by gunnarmorling** [comment]
Applied. Thx, @jpechane!

**Pull request merged**